### DISCUSSION TITLE
[Documentation] memory arena, reuse, and pattern

### DISCUSSION THREAD

**Discussion by mohsenMahmoodzadeh** [microsoft/onnxruntime#13409] [Other Q&A]
As described in Python API Doc, there are some params in session options coressponding to memory configuration such as:

- enable_cpu_mem_arena
- enable_mem_usage
- enable_mem_pattern

There are some descriptions for them tough, but I can not understaned their usage and the concepts behind them precisely.

Can anyone give me some clarity about these parameters?

**Comment by faxu** [comment]
Mem arena: https://github.com/microsoft/onnxruntime/blob/7d5833229877ad09265a75983d005d0f39457209/onnxruntime/core/framework/session_options.h#L75

Memory reuse: https://github.com/microsoft/onnxruntime/blob/7d5833229877ad09265a75983d005d0f39457209/onnxruntime/core/framework/session_options.h#L70

Memory pattern: https://github.com/microsoft/onnxruntime/blob/7d5833229877ad09265a75983d005d0f39457209/onnxruntime/core/framework/session_options.h#L63

More reading on memory management: 
* https://onnxruntime.ai/docs/get-started/with-c.html#features
* https://onnxruntime.ai/docs/performance/tune-performance.html#tips-for-tuning-performance

**Comment by mohsenMahmoodzadeh** [comment]
Thank you for your responses. But I have some problems with ```Memory Reuse``` and ```Memory Pattern Optimization```. 

Can you give some conceptual background about what they are and what they do? I read the references you mentioned but couldn't understand precisely their usages.
